services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankealueService.kt

```diff
@@ -30,7 +30,15 @@ class HankealueService(
     /** Map by area geometry id to area geometry data. */
     fun geometryMapFrom(alueet: List<HankealueEntity>): Map<Int, Geometriat?> =
         alueet
-            .mapNotNull { it.geometriat }
+            .mapNotNull {
+                // If car traffic nuisance indexes are not yet calculated, calculate them now.
+                it.tormaystarkasteluTulos?.let { tt ->
+                    if (tt.haitanKesto == -1) {
+                        updateTormaystarkastelu(it)
+                    }
+                }
+                it.geometriat
+            }
```

Comment:
I can see what this is trying to do, but I think this is a lousy way of handling the data migration.

* We can't know if the migration will ever be completed. Someone needs to access the hanke to update its indices. For some draft hanke, this might never happen.
* This needs to use a hacky invalid value for `haitanKesto`.
* We're changing data in a mapper that's supposed to be a read-only method.
  - This is potentially called outside any transactions and the like.
  - This will make HTTP GET calls to cause data changes.

Since we have no indices calculated in production, we could delete all indices from the DB. They would be recalculated when someone edits the hanke. We might have to mark all hanke as drafts as well.

It should also be possible to recalculate the indices completely in the SQL migration, but that might be a painful SQL script to write and verify.

Reply:
Yes, you're right. I will rewrite the database migration to delete all indices. I can also look into the SQL way of calculating the indices in the migration script but as you said, it would be a not-too-easy task.

Reply:
I did not go for the SQL migration scripting but just simply delete all `tormaystarkastelutulos` rows and set project status to `DRAFT`.